pkg/platform/interface.go

```diff
@@ -20,6 +21,7 @@ type TestReportDB interface {
 	GetResults(runId string) ([]KindSpecifier, error)
 	Read(ctx context.Context, path, name string) (KindSpecifier, error)
 	Write(ctx context.Context, path string, doc KindSpecifier) error
+	ReadReportSessionIndices(path string) ([]string, error)
```

Comment:
incase of reports there's no path defined so yes needed, But removing ReadReportSessionIndices as it isn't used anywhere.

Reply:
If ReadReportSessionIndices is not present, how are we going to name the test report? Doesn't the constructor of the report's DB doesn't have path in it..?

Reply:
testReportPath is defined in test.go file and passed on everywhere. testReportPath then used findLastIndex to create the report name

Reply:
You might have understood by now that if we can use ReadReportSessionIndices and include path in the constructor and remove path from other functions, this will make it clean but I am leaving this to you if you want to do this in this pr or not. You can resolve the conversation if you want to skip it for now.

Reply:
the path thing can be fixed in other PR as this change won't align with goal of this PR